### PR TITLE
Update the plugin to the new logtash-core-plugin-api

### PR DESCRIPTION
**Pull request by ph**
Ref: https://github.com/elastic/logstash/issues/5269

### CODE REVIEW

File: logstash-input-example.gemspec [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Gem::Specification.new do |s|
   s.name = 'logstash-input-example'
-  s.version         = '2.0.4'
+  s.version = '1.0.0'
```
Comment (purbon) [inline logstash-input-example.gemspec]: why downgrade version?

Comment (ph) [inline logstash-input-example.gemspec]: I thought this plugin wasn't publishing on rubygems but this isnt the case.

Comment (ph) [inline logstash-input-example.gemspec]: I will revert that.

**Comment by ph** [comment]
Update from review.

**Comment by purbon** [comment]
LGTM

**Comment by ph** [comment]
@purbon I took another look at the dependency of this PR and sadly they were wrong we are now using a range on the `logstash-core-plugin-api` gem, I've updated this PR with the appropriate version.

**Comment by ph** [comment]
I believe the version of this gem should be 1.0.0, I mean as a plugin author I would like to start with this version and not have to replace it.

I think the version changes shouldn't matter in this plugin since we shouldn't publish them.

**Comment by ph** [comment]
Also, I've pushed a new version of the constraint for this plugin and fixed the tests. 
The plugin was missing a runtime dependency on the `logstash-codec-plain`, Its been like this for some time.

**Comment by purbon** [comment]
```
LoadError: JRuby ext built for wrong Java version in `JrubyEventExtService': java.lang.UnsupportedClassVersionError: JrubyEventExtService : Unsupported major.minor version 52.0
  require at org/jruby/RubyKernel.java:1040
   (root) at /home/travis/build/logstash-plugins/logstash-input-example/vendor/bundle/jruby/1.9/gems/logstash-core-event-java-5.0.0.alpha5.snapshot1-java/lib/logstash-core-event-java/logstash-core-event-java.rb:25
  require at org/jruby/RubyKernel.java:1040
   (root) at /home/travis/build/logstash-plugins/logstash-input-example/vendor/bundle/jruby/1.9/gems/logstash-core-event-java-5.0.0.alpha5.snapshot1-java/lib/logstash-core-event.rb:1
```

wrong java version used for travis tests?

**Comment by ph** [comment]
@purbon travis is now green.

**Pull request merged**